### PR TITLE
feat(new): Added Azure.Cosmos.ContinuousBackup

### LINKED ISSUE #2954: [RULE] CosmosDB continuous backup mode
### Existing rule

_No response_

### Suggested rule

As per [APRL recommendation](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/azure-resources/DocumentDB/databaseAccounts/#configure-continuous-backup-mode), configuring Cosmos DB's backup to always on, offers protection against data mishaps. Continuous mode allows for self-serve restoration to a pre-mishap point, unlike periodic mode which requires contacting Microsoft support, leading to longer restore times.

Resource Type: Microsoft.DocumentDb/databaseAccounts

### Pillar

Reliability

### Additional context

[Continuous backup with point in time restore feature in Azure Cosmos DB | Microsoft Learn](https://learn.microsoft.com/azure/cosmos-db/continuous-backup-restore-introduction)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2954

Added Azure.Cosmos.ContinuousBackup.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Ready for review @BernieWhite.

File: src/PSRule.Rules.Azure/rules/Azure.Cosmos.Rule.yaml [hunk 1]
```diff
@@ -108,4 +108,22 @@ spec:
...
+    ruleSet: 2024_09
+    Azure.WAF/pillar: Reliability
+spec:
+  type:
+    - Microsoft.DocumentDb/databaseAccounts
+  condition:
+    field: properties.backupPolicy.type
+    equals: Continuous
+
 #endregion Rules
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.Cosmos.Rule.yaml:129]: Run for APIs except Cassandra.

```suggestion
---
# Synopsis: Cosmos DB has continuous backup configured.
apiVersion: github.com/microsoft/PSRule/v1
kind: Rule
metadata:
  name: Azure.Cosmos.ContinuousBackup
  ref: AZR-000439
  tags:
    release: GA
    ruleSet: 2024_09
    Azure.WAF/pillar: Reliability
spec:
  type:
    - Microsoft.DocumentDb/databaseAccounts
  with:
    - Azure.Cosmos.IsMongo
    - Azure.Cosmos.IsGremlin
    - Azure.Cosmos.IsTable
    - Azure.Cosmos.IsNoSQL
  condition:
    allOf:
    - field: properties.backupPolicy.type
      equals: Continuous
    - field: properties.backupPolicy.continuousModeProperties.tier
      hasValue: true

#endregion Rules

#region Selectors

---
# Synopsis: Cosmos DB accounts that use the Mongo API.
apiVersion: github.com/microsoft/PSRule/v1
kind: Selector
metadata:
  name: Azure.Cosmos.IsMongo
spec:
  if:
    field: properties.capabilities[*]
    allOf:
      - field: name
        equals: EnableMongo
    greaterOrEqual: 1

---
# Synopsis: Cosmos DB accounts that use the Cassandra API.
apiVersion: github.com/microsoft/PSRule/v1
kind: Selector
metadata:
  name: Azure.Cosmos.IsCassandra
spec:
  if:
    field: properties.capabilities[*]
    allOf:
      - field: name
        equals: EnableCassandra
    greaterOrEqual: 1

---
# Synopsis: Cosmos DB accounts that use the Gremlin API.
apiVersion: github.com/microsoft/PSRule/v1
kind: Selector
metadata:
  name: Azure.Cosmos.IsGremlin
spec:
  if:
    field: properties.capabilities[*]
    allOf:
      - field: name
        equals: EnableGremlin
    greaterOrEqual: 1

---
# Synopsis: Cosmos DB accounts that use the Table API.
apiVersion: github.com/microsoft/PSRule/v1
kind: Selector
metadata:
  name: Azure.Cosmos.IsTable
spec:
  if:
    field: properties.capabilities[*]
    allOf:
      - field: name
        equals: EnableTable
    greaterOrEqual: 1

---
# Synopsis: Cosmos DB accounts that use the NoSQL API.
apiVersion: github.com/microsoft/PSRule/v1
kind: Selector
metadata:
  name: Azure.Cosmos.IsNoSQL
spec:
  if:
    field: properties.capabilities[*]
    allOf:
      - field: name
        in:
          - EnableMongo
          - EnableCassandra
          - EnableTable
          - EnableGremlin
    count: 0

#endregion Selectors
```

**Comment by BenjaminEngeset** [comment]
Thanks for all the great feedback @BernieWhite. I have adjusted accordingly now. Do you mind taking a look?

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**